### PR TITLE
feat: add publish handler

### LINKED ISSUE #52: Simple event publisher
Similar to the [CF worker](https://github.com/getAlby/http-nostr-publisher) add an option to publish an event to one or multiple (max 5) relays.

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #52

### CODE REVIEW

File: internal/nostr/nostr.go [hunk 1]
```diff
@@ -203,6 +204,67 @@ func (svc *Service) InfoHandler(c echo.Context) error {
...
+	if (requestData.SignedEvent == nil) {
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "signed event is empty",
+		})
+	}
+
+	var wg sync.WaitGroup
+	errorCh := make(chan error, len(requestData.Relays))
+
+	for _, relayUrl := range requestData.Relays {
```
Comment (bumi) [inline internal/nostr/nostr.go:230]: let's support here only one relay (as in all other endpoints I think) and if no relay is given we use the default realy.

Comment (im-adithya) [reply internal/nostr/nostr.go]: Changed to support only one relay now!

**Comment by im-adithya** [comment]
⚠️ Have to add README

File: internal/nostr/nostr.go [hunk 2]
```diff
@@ -203,6 +203,59 @@ func (svc *Service) InfoHandler(c echo.Context) error {
...
+	err = relay.Publish(ctx, *requestData.SignedEvent)
+	if err != nil {
+		svc.Logger.WithError(err).WithFields(logrus.Fields{
+			"eventId":   requestData.SignedEvent.ID,
+			"relayUrl":  requestData.RelayUrl,
+		}).Error("Failed to publish event")
+	
+		return c.JSON(http.StatusInternalServerError, ErrorResponse{
+			Message: "error publishing the event: " + err.Error(),
+		})
```
Comment (bumi) [inline internal/nostr/nostr.go:248]: can we provide some more details here? 
what are good status codes and what is a useful response JSON for the API consumer/developer.

File: internal/nostr/nostr.go [hunk 3]
```diff
@@ -203,6 +203,59 @@ func (svc *Service) InfoHandler(c echo.Context) error {
...
+	}
+
+	svc.Logger.WithFields(logrus.Fields{
+		"eventId":   requestData.SignedEvent.ID,
+		"relayUrl":  requestData.RelayUrl,
+	}).Info("Published event")
+
+	return c.JSON(http.StatusOK, "published")
```
Comment (bumi) [inline internal/nostr/nostr.go:256]: be consistent with the API response. 
always return JSON, always try to return something useful.

File: internal/nostr/nostr.go (hunk 2 above)
Comment (im-adithya) [reply internal/nostr/nostr.go]: Modified ErrorResponse type to always have error, so users can check for resp.error and confirm if the result is successful

```golang
type ErrorResponse struct {
	Message string `json:"message"`
	Error   string `json:"error"`
}
```

Only using `StatusBadRequest 400`, `StatusRequestTimeout 408` and `StatusInternalServerError 500`

Comment (bumi) [reply internal/nostr/nostr.go]: great. 
maybe also compare it to the other APIs we have - can we take some inspiration from there (though we are not really good there, too)

**Pull request merged**